app/experimenter/experiments/serializers.py

```diff
@@ -616,7 +616,36 @@ def update(self, instance, validated_data):
         return instance
 
 
-class ExperimentDesignPrefSerializer(ExperimentDesignBaseSerializer):
+class PrefValidationMixin(serializers.ModelSerializer):
+
+    def validate_pref(self, pref_type, pref_value, error_list):
+        if pref_type == "integer":
+            try:
+                int(pref_value)
+            except ValueError:
+                error_list.append({"value": "The pref value must be an integer."})
+            else:
+                error_list.append({})
+
+        if pref_type == "boolean":
+            if pref_value not in ["true", "false"]:
+                error_list.append({"value": "The pref value must be a boolean."})
+
+            else:
+                error_list.append({})
+
+        if pref_type == "json string":
+            try:
+                json.loads(pref_value)
+            except ValueError:
+                error_list.append({"value": "The pref value must be valid JSON."})
+            else:
+                error_list.append({})
+        if pref_type == "string":
+            error_list.append({})
```

Comment:
Do we need this at all?

Reply:
OH I see.  You're passing in the list, no this is too specific to `variants`, just build a dictionary in here and return it and then append to `error_list` outside.